extensions/wikia/Chat2/ChatController.class.php

```diff
@@ -94,6 +117,44 @@ public function executeIndex() {
 	}
 
 	/**
+	 * Add MediaWiki:Chat.js and MediaWiki:Chat.css to output, respecting configuration
+	 * @param OutputPage $out
+	 */
+	private function addSiteJsCss( OutputPage $out ) {
+		switch ( true ) {
+			case ( $this->wg->UseSiteJs && $this->wg->UseSiteCss ):
+				$out->addModules( 'chat.site' );
+				break;
+			case $this->wg->UseSiteJs:
+				$out->addModuleScripts( 'chat.site' );
+				break;
+			case $this->wg->UseSiteCss:
+				$out->addModuleStyles( 'chat.site' );
+				break;
+			default:
+		}
+	}
+
+	/**
+	 * Add User:UserName/chat.js and User:UserName/chat.css to output, respecting configuration
+	 * @param OutputPage $out
+	 */
+	private function addUserJsCss( OutputPage $out ) {
+		switch ( true ) {
+			case ( $this->wg->AllowUserJs && $this->wg->AllowUserCss ):
+				$out->addModules( 'chat.user' );
+				break;
+			case $this->wg->AllowUserJs:
+				$out->addModuleScripts( 'chat.user' );
+				break;
+			case $this->wg->AllowUserCss:
+				$out->addModuleStyles( 'chat.user' );
+				break;
+			default:
+		}
+	}
+
+	/**
 	 * adding js variable
 	 */
 	public function onMakeGlobalVariablesScript( array &$vars ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Don't use an OnMakeGlobalVariablesScript hook handler to register JS variables. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AOnMakeGlobalVariablesScript)